### ISSUE TITLE
Caret position is not preserved after "Move to Other View"

### ISSUE THREAD

**Issue by Yaron10** [notepad-plus-plus/notepad-plus-plus#2541]
Hello,

STR:
Have two files open in Single-View mode.
Move to Other View.
Move the caret in both files.
Move to Other View (the one moved earlier).

Result:
The moved file caret position is preserved (correct).
The caret in the other file is back to its pre-Move position (wrong).
***
Possible solution:
In Notepad_plus.cpp replace

```
else    //open the document, also copying the position
    {
        loadBufferIntoView(current, viewToGo);
        Buffer *buf = MainFileManager->getBufferByID(current);
        _pEditView->saveCurrentPos();   //allow copying of position
        buf->setPosition(buf->getPosition(_pEditView), _pNonEditView);
        _pNonEditView->restoreCurrentPos(); //set position
        activateBuffer(current, viewToGo);
    }
```
with

```
	else	//open the document, also copying the position
	{
		bool modeDoubleView = _pEditView->isVisible() && _pNonEditView->isVisible();
		if (modeDoubleView)
		{
			BufferID currentUnfocused = _pNonEditView->getCurrentBufferID();	// Save position of currentUnfocused.
			Buffer *bufUnfocused = MainFileManager->getBufferByID(currentUnfocused);
			_pNonEditView->saveCurrentPos();
			bufUnfocused->setPosition(bufUnfocused->getPosition(_pNonEditView), _pEditView);
		}

		loadBufferIntoView(current, viewToGo);
		Buffer *buf = MainFileManager->getBufferByID(current);
		_pEditView->saveCurrentPos();	//allow copying of position
		buf->setPosition(buf->getPosition(_pEditView), _pNonEditView);
		_pNonEditView->restoreCurrentPos();	//set position

		if (modeDoubleView)	    // Restore position of currentUnfocused.
			_pEditView->restoreCurrentPos();

		activateBuffer(current, viewToGo);
	}
```
It doesn't deal with the root of the problem but it solves the issue.

Thank you.

**Comment by Yaron10** [comment]
Hello @SinghRajenM,

@pnedev has been doing wonders to the most popular [Compare Plugin](https://github.com/pnedev/compare-plugin).

This issue and https://github.com/notepad-plus-plus/notepad-plus-plus/issues/2277 affect Compare Plugin to some degree and should be addressed in NPP.

May I ask you to preview the proposed solutions and submit them as PRs?
I would highly appreciate that.

NOTE:
If you do take it, the code proposed here can be slightly optimized by modifying the beginning of the function.

Thank you very much.
Best regards.

**Comment by SinghRajenM** [comment]
@Yaron10 @pnedev 
I've submitted fix #2664 for this issue. I'll check for #2277 .
Thank you!
